src/NDItk/multigroup/ScatteringMatrix/src/read.hpp

```diff
@@ -0,0 +1,39 @@
+/**
+ *  @brief Read the record data
+ *
+ *  @param[in,out] iter        an iterator to the current position in the input
+ *  @param[in,out] end         an iterator to the end of the input
+ *  @param[in,out] incident    the number of incident energy groups
+ *  @param[in,out] outgoing    the number of outgoing energy groups
+ *  @param[in,out] number      the number of moments
+ */
+template< typename Iterator >
+void read( Iterator& iter, const Iterator& end, unsigned int incident,
+           unsigned int outgoing, unsigned int number ) {
+
+  // read in the data
+  this->incident_ = incident;
+  this->outgoing_ = outgoing;
+  this->number_moments_ = number;
+  base::RealListRecord::read( iter, end, number * ( 1 + incident * outgoing ) );
+  this->generateBlocks();
+
+  // verify
+  verify( this->moments() );
+};
+
+/**
+ *  @brief Read the record data
+ *
+ *  @param[in,out] iter        an iterator to the current position in the input
+ *  @param[in,out] end         an iterator to the end of the input
+ *  @param[in,out] incident    the number of incident energy groups
+ *  @param[in,out] outgoing    the number of outgoing energy groups
```

Comment:
This one doesn't have outgoing. Also, can you put in the brief the difference between these two reads?

Reply:
Yes, the first one is basically for reading a record that has matrices of size m * n with m != n (basically the outgoing particle production matrices) while the second one is for an m * m matrix (thus the scattering matrix). The second one does use the first one under the hood.

I've updated it and added it to fix/review-part5